runelite-client/src/main/java/net/runelite/client/plugins/xpglobes/XpGlobesOverlay.java

```diff
@@ -54,84 +54,109 @@
 @Slf4j
 public class XpGlobesOverlay extends Overlay
 {
-	private final Client client;
-	private final XpGlobesPlugin plugin;
-	private final XpGlobesConfig config;
-	private final XpTrackerService xpTrackerService;
-	private final PanelComponent xpTooltip = new PanelComponent();
-
-	@Inject
-	private SkillIconManager iconManager;
-
 	private static final int MINIMUM_STEP = 10;
-
 	private static final int PROGRESS_RADIUS_START = 90;
 	private static final int PROGRESS_RADIUS_REMAINDER = 0;
-
 	private static final int DEFAULT_START_Y = 10;
-
 	private static final int TOOLTIP_RECT_SIZE_X = 150;
 
+	private final Client client;
+	private final XpGlobesPlugin plugin;
+	private final XpGlobesConfig config;
+	private final XpTrackerService xpTrackerService;
+	private final PanelComponent xpTooltip = new PanelComponent();
+	private final SkillIconManager iconManager;
+
 	@Inject
-	public XpGlobesOverlay(Client client, XpGlobesPlugin plugin, XpGlobesConfig config, XpTrackerService xpTrackerService)
+	private XpGlobesOverlay(
+		Client client,
+		XpGlobesPlugin plugin,
+		XpGlobesConfig config,
+		XpTrackerService xpTrackerService,
+		SkillIconManager iconManager)
 	{
-		setPosition(OverlayPosition.DYNAMIC);
-		setPriority(OverlayPriority.HIGH);
+		this.iconManager = iconManager;
 		this.client = client;
 		this.plugin = plugin;
 		this.config = config;
 		this.xpTrackerService = xpTrackerService;
+		setPosition(OverlayPosition.DETACHED);
+		setPriority(OverlayPriority.HIGH);
 	}
 
 	@Override
-	public Dimension render(Graphics2D graphics)
+	public Rectangle getBounds()
 	{
 		//if this is null there is no reason to draw e.g. switching between resizable and fixed
-		Widget viewportWidget = client.getViewportWidget();
+		final Widget viewportWidget = client.getViewportWidget();
 		if (viewportWidget == null)
 		{
-			return null;
+			return new Rectangle();
 		}
 
-		//check the width of the client if we can draw properly
-		int clientWidth;
-		switch (config.centerOrbs())
+		final int queueSize = plugin.getXpGlobesSize();
+		if (queueSize == 0)
 		{
-			case MIDDLE_CANVAS:
-				clientWidth = client.getViewportWidth();
-				break;
-			case MIDDLE_VIEWPORT:
-				clientWidth = viewportWidget.getWidth();
-				break;
-			case DYNAMIC:
-				clientWidth = (viewportWidget.getWidth() + client.getViewportWidth()) / 2;
-				break;
-			default:
-				clientWidth = client.getViewportWidth();
-				break;
+			return new Rectangle();
 		}
-		if (clientWidth <= 0)
+
+		// Get width of markers
+		final int markersLength = (queueSize * (config.xpOrbSize())) + ((MINIMUM_STEP) * (queueSize - 1));
+
+		final int startDrawX;
+		final int startDrawY;
+		if (getPreferredLocation() == null)
 		{
-			return null;
+			//check the width of the client if we can draw properly
+			final int clientWidth;
+			switch (config.centerOrbs())
+			{
+				case MIDDLE_CANVAS:
```

Comment:
This seems to share code with the default--if that's the plan going forward for this configuration, remove the code and tack it onto the default case:

```
case MIDDLE_CANVAS:
default:
    ...
```

Reply:
Well i just copy-pastad the code so idc mostly about it. I can clean it up later, the entire plugin is mess atm anyway